src/contracts/pods/EigenPodManager.sol

```diff
@@ -642,23 +643,13 @@ contract EigenPodManager is
         ownerToPod[podOwner].withdrawRestakedBeaconChainETH(recipient, amount);
     }
 
-    /**
-     * @notice If the `staker` has no existing shares, then they can `undelegate` themselves.
-     * This allows people a "hard reset" in their relationship with EigenLayer after withdrawing all of their stake.
-     * @param staker The address to undelegate. Passed on as an input to the `delegationManager.undelegate` function.
-     */
-    function _undelegate(address staker) internal onlyNotFrozen(staker) {
-        require(stakerCanUndelegate(staker), "EigenPodManager._undelegate: staker has active deposits");
-        delegationManager.undelegate(staker);
-    }
-
     /**
      * @notice Internal function to enter `podOwner` into undelegation limbo
      * @dev Does nothing if the `podOwner` has no delegated shares (i.e. they are already in undelegation limbo or have no shares)
      * OR if they are not actively delegated to any operator.
      */
     function _enterUndelegationLimbo(address podOwner) internal {
-        if (!stakerHasNoDelegatedShares(podOwner) && delegationManager.isDelegated(podOwner)) {
+        if (!podOwnerHasNoDelegatedShares(podOwner) && delegationManager.isDelegated(podOwner)) {
```

Comment:
Not a fan of the additional dependencies here - feels like implementing delegationmgr logic in epmgr

Reply:
do you think it would make more sense for entering this "undelegation limbo" mode to always originate from a call to the DelegationManager? then this flow would probably be more logical / checks could happen more "in place"?

Reply:
yes, that's what i've spent some time implementing locally! i got sidetracked by that bug in withdrawal queues, but my initial findings were that things were much more logically separated.

Reply:
> yes, that's what i've spent some time implementing locally! i got sidetracked by that bug in withdrawal queues, but my initial findings were that things were much more logically separated.

makes sense! if you're implementing it yourself, I can avoid replicating your work? feel free to push when ready and I can review? Or let me know if you prefer I implement it myself and we can compare.

Reply:
addressed as part of PR https://github.com/Layr-Labs/eigenlayer-contracts/pull/183 -- or see the merge commit https://github.com/Layr-Labs/eigenlayer-contracts/pull/176/commits/75e1c96437486846dce1b391fe59c71a88598113